File: 588/b.cpp
#include <bits/stdc++.h>

using namespace std;

#define _ ios_base::sync_with_stdio(0);cin.tie(0);
#define endl '\n'
#define f first
#define s second
#define pb push_back

typedef long long ll;
typedef pair<int, int> ii;

const int INF = 0x3f3f3f3f;
const ll LINF = 0x3f3f3f3f3f3f3f3fll;

int main(){ _
	int n, k;
	cin >> n >> k;
	string s;
	cin >> s;
	int k_aux = 0;
	int i = 0;
	if (i == 0 and k >0) {
		if (s[i] == '1')
			i++;
		else {
			s[i] = '1';
			k_aux++;
			i++;
		}
	}
	while (i < n and k_aux < k){
		if (s[i] == '0')
			i++;
		else {
			s[i] = '0';
			k_aux++;
			i++;
		}
	}

	if (n==1 and k >0)
		cout << 0 << endl;
	else
		cout << s << endl;

	exit(0);
}

File: 588/B.py
n, k = map(int, input().split())

s = input()

k_aux = 0
i = 0
if i == 0 and k > 0:
	if s[i] == '1':
		i += 1
	else:
		s =  s[:i] + '1' + s[i+1:]
		k_aux += 1
		i += 1
while i < n and k_aux < k:
	if k-k_aux >= n-i:
		s = s[:i] + '0'*(n-i)
		i += n-i
		k_aux += n-i
		continue
	elif s[i] == '0':
		i += 1
		continue
	else:
		s = s[:i] + '0' + s[i+1:]
		k_aux += 1
	i += 1

if n == 1 and k > 0:
	print(0)
else:
	print(s)
